libs/brain/switch/src/lib/brn-switch.component.ts

```diff
@@ -239,30 +251,59 @@ export class BrnSwitchComponent implements AfterContentInit, OnDestroy {
 		this.switch().nativeElement.dispatchEvent(new Event('change'));
 	}
 
-	ngOnDestroy() {
+	public ngOnDestroy() {
 		this._focusMonitor.stopMonitoring(this._elementRef);
 	}
 
-	/** We intercept the id passed to the wrapper component and pass it to the underlying button switch control **/
+	/**
+	 * Gets proper ID for inner button element.
+	 * Removes '-switch' suffix if present in container ID.
+	 *
+	 * @param idPassedToContainer - ID applied to container element
+	 * @returns ID to use for inner button or null
+	 */
 	protected getSwitchButtonId(idPassedToContainer: string | null | undefined): string | null {
 		return idPassedToContainer ? idPassedToContainer.replace(CONTAINER_POST_FIX, '') : null;
 	}
 
-	writeValue(value: boolean): void {
+	/**
+	 * Updates internal state when control value changes from outside.
+	 * Part of ControlValueAccessor interface.
+	 *
+	 * @param value - New checked state
+	 */
+	public writeValue(value: boolean): void {
 		this.checked.set(Boolean(value));
 	}
 
-	registerOnChange(fn: ChangeFn<boolean>): void {
+	/**
+	 * Registers callback for value changes.
+	 * Part of ControlValueAccessor interface.
+	 *
+	 * @param fn - Function to call when value changes
+	 */
+	public registerOnChange(fn: ChangeFn<boolean>): void {
 		this._onChange = fn;
 	}
 
-	registerOnTouched(fn: TouchFn): void {
+	/**
+	 * Registers callback for touched events.
+	 * Part of ControlValueAccessor interface.
+	 *
+	 * @param fn - Function to call when control is touched
+	 */
+	public registerOnTouched(fn: TouchFn): void {
 		this._onTouched = fn;
 	}
 
-	/** Implemented as a part of ControlValueAccessor. */
-	setDisabledState(isDisabled: boolean): void {
+	/**
+	 * Updates disabled state from form control.
+	 * Part of ControlValueAccessor interface.
+	 *
+	 * @param isDisabled - Whether switch should be disabled
+	 */
+	public setDisabledState = (isDisabled: boolean): void => {
```

Comment:
Just curious - was there a reason this needed to be converted to a class property instead of a method?